tests/multi_modal_search.test.ts

```diff
@@ -0,0 +1,167 @@
+    //     ],
+    //   },
+    // },
+  },
+  searchFragments: {
+    // poster: {
+    //   value: {
+    //     content: [
+    //       {
+    //         type: "image_url",
+    //         image_url: "{{media.poster}}",
+    //       },
+    //     ],
+    //   },
+    // },
+    // text: {
+    //   value: {
+    //     content: [
+    //       {
+    //         type: "text",
+    //         // uses the `q` field from search queries
+    //         text: "{{q}}",
+    //       },
+    //     ],
+    //   },
+    // },
+    textAndPoster: {
+      value: {
+        content: [
+          {
+            type: "text",
+            text: "{{q}}",
+          },
+          {
+            type: "image_url",
+            image_url: "{{media.poster}}",
+          },
+        ],
+      },
+    },
+  },
+  request: {
+    inputs: ["{{fragment}}", "{{..}}"],
+    model: "voyage-multimodal-3",
+  },
+  response: {
+    // Maps how Voyage API returns the embedding
+    data: [
+      {
+        embedding: "{{embedding}}",
+      },
+      "{{..}}",
+    ],
+  },
+} satisfies Embedder;
+
+describe.skipIf(!VOYAGE_API_KEY)("Multi-modal search", () => {
+  let searchClient: Meilisearch;
+
+  beforeAll(async () => {
+    const client = await getClient("Admin");
+    await client.updateExperimentalFeatures({
+      multimodal: true,
+    });
+    await client.createIndex(INDEX_UID).waitTask();
+    await client.index(INDEX_UID).updateSettings({
+      searchableAttributes: ["title", "overview"],
+      embedders: {
+        [EMBEDDER_NAME]: EMBEDDER_CONFIG,
+      },
+    });
+    await client.index(INDEX_UID).addDocuments(movies).waitTask();
+    searchClient = await getClient("Search");
+  });
+
+  test("should work with text query", async () => {
+    const response = await searchClient
+      .index(INDEX_UID)
+      .search("A movie with lightsabers in space", {
+        hybrid: {
+          embedder: EMBEDDER_NAME,
+          semanticRatio: 1,
+        },
+      });
+    expect(response.hits[0].title).toBe("Star Wars");
+  });
+
+  test("should work with image query", async () => {
+    const theFifthElementPoster = movies[3].poster;
+
+    const response = await searchClient.index(INDEX_UID).search(null, {
+      media: {
+        poster: theFifthElementPoster,
+      },
+      hybrid: {
+        embedder: EMBEDDER_NAME,
+        semanticRatio: 1,
+      },
+    });
+    expect(response.hits[0].title).toBe("The Fifth Element");
+  });
+
+  test("should work with text and image query", async () => {
+    const spaceImageUrl =
+      "https://science.nasa.gov/wp-content/uploads/2023/06/webb-flickr-52259221868-30e1c78f0c-4k-jpg.webp";
+    const response = await searchClient.index(INDEX_UID).search(null, {
+      q: "A futuristic movie",
+      media: {
+        poster: spaceImageUrl,
+      },
+      hybrid: {
+        embedder: EMBEDDER_NAME,
+        semanticRatio: 1,
+      },
+    });
+    expect(response.hits[0].title).toBe("Star Wars");
+  });
```

Comment:
When using `textAndPoster` and removing the others (`text` and `poster`) fragments, only this test passes.